nilearn/signal.py

```diff
@@ -351,18 +368,31 @@ def butterworth(signals, sampling_rate, low_pass=None, high_pass=None,
             warnings.warn(
                 'Signals are returned unfiltered because band-pass critical '
                 'frequencies are equal. Please check that inputs for '
-                'sampling_rate, low_pass, and high_pass are valid.')
+                'sampling_rate, low_pass, and high_pass are valid.'
+            )
             if copy:
                 return signals.copy()
             else:
                 return signals
     else:
         critical_freq = critical_freq[0]
 
-    b, a = sp_signal.butter(order, critical_freq, btype=btype, output='ba')
+    b, a = sp_signal.butter(
+        order,
+        critical_freq,
+        btype=btype,
+        output='ba',
+        fs=sampling_rate,
```

Comment:
This parameter changes the results, and I don't know why. I'm not sure if including the sampling rate fixes a bug or introduces one.

Reply:
My guess is that this is why tests are failing as well.

Reply:
I think I understand what `fs` does now. Without `fs`, we need to divide the frequencies by the nyquist frequency. With `fs`, we can just feed in the frequencies. I think the code is more interpretable if we include `fs` and _don't_ modify the frequencies, but it's also primarily a stylistic change. @bthirion do you have an opinion on this?

Reply:
I think that people are used to thinking in terms of absolute frequencies.  So I would enforce the use of fs.

Reply:
Sounds good! I'll update the code.